### ISSUE TITLE
False Positive from match_str_case_mismatch

### ISSUE THREAD

**Issue by MinerSebas** [rust-lang/rust-clippy#7882]
Lint name: match_str_case_mismatch

I tried this code:

```rust
	match "DX12".to_lowercase().as_str() {
        "dx12" => (),
        _ => (),
    }
```

I expected to see this happen: The Lint doesn't Trigger.

Instead, this happened: The Lint triggered:
```
error: this `match` arm has a differing case than its expression
 --> src\main.rs:3:9
  |
3 |         "dx12" => (),
  |         ^^^^^^
  |
  = note: `#[deny(clippy::match_str_case_mismatch)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_str_case_mismatch
help: consider changing the case of this arm to respect `to_lower_case`
  |
3 |         "dx12" => (),
  |         ~~~~~~
```

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.58.0-nightly (00d5e42e7 2021-10-24)
binary: rustc
commit-hash: 00d5e42e776da900049fe19087bc9b0057ec70cd
commit-date: 2021-10-24
host: x86_64-pc-windows-msvc
release: 1.58.0-nightly
LLVM version: 13.0.0
```
